plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -428,20 +413,39 @@ export class GitLabClient {
       }
     }
 
-    const response = await fetch(
-      request.toString(),
-      getGitLabRequestOptions(this.config),
-    );
-
-    if (!response.ok) {
-      throw new Error(
-        `Unexpected response when fetching ${request.toString()}. Expected 200 but got ${
-          response.status
-        } - ${response.statusText}`,
+    for (let i = 0; ; i++) {
+      this.logger.debug(`Fetching: ${request.toString()}`);
+      const response = await fetch(
+        request.toString(),
+        getGitLabRequestOptions(this.config),
       );
-    }
 
-    return response.json();
+      if (response.status === tooManyRequests && i < maxRetry) {
+        const retryAfter = response.headers.get('Retry-After');
+
+        if (retryAfter) {
+          const delay = isNaN(Number(retryAfter))
```

Comment:
mmm I guess some tests are missing here, as `delay` is currently a boolean

Reply:
actually, it's either a Date or a number; the `boolean` was just the intro to a ternary expression 😄

Reply:
sorry I'm stupid, looks good 👍